### ISSUE TITLE
Ratings Calculations Slightly Off

### ISSUE THREAD

**Issue by Diet6** [mynian/ConvertRatings#34]
I've noticed on my priest (not sure if this applies to other classes) that my ratings are a bit off when I compare the tooltip to my stat sheet and remove/swap the item.

Here's my legs for example:
![image](https://user-images.githubusercontent.com/111312010/184706810-ab020851-d1b3-41c7-b80f-ae81f318cfae.png)
And my stat sheet (using deja character stats):
![image](https://user-images.githubusercontent.com/111312010/184706965-137865f0-fa6a-4a7c-adcf-8a3196efe19b.png)

And then here's my stat sheet with the legs removed:
![image](https://user-images.githubusercontent.com/111312010/184707039-bd8bef2a-0b4b-4c5a-b206-9ce6a2c2408d.png)

The crit from the legs said it was worth 1.91%, but removing them dropped me 1.94%.
Similarly, the mastery was listed as worth 4.31%, but removing them dropped me 4.40%

I also see the same thing with stats like leech:
![image](https://user-images.githubusercontent.com/111312010/184707573-da8f4f47-9e73-43a7-97cc-53a3a45158d8.png)
![image](https://user-images.githubusercontent.com/111312010/184707601-e6b7f755-bb66-407b-9d3e-a85fa361b230.png)
Then trinket removed:
![image](https://user-images.githubusercontent.com/111312010/184707653-13d17460-211b-41a9-9132-f41d16cb8a19.png)

It went from 14% to 11.79%, but if the trinket was worth 2.76% it should have dropped to 11.24%.

This applies to all stats, including swapping gear (though the numbers get a bit harder to follow so I went with removing gear).

Maybe this is an artifact of the diminishing returns? Though I believe I was well below the cap on the crit at least.

Let me know if any other information would be helpful!

**Comment by mynian** [comment]
I'll be honest, it probably does have something to do with how DR is calculated. I haven't looked at how DR is calculated since it was originally introduced and I'm not 100% positive if my way of calculating it is perfectly accurate. I wrote the code for calculating DR with no way to test it as I don't really have any way to force my stats to the different DR thresholds.

**Comment by mynian** [comment]
So i think I actually figured this one out. I was playing around with some gear and noticed a difference on stuff that shouldn't have been affected by DRs. I took a look at the stat scaling data and apparently at some point from my initial release with the data for levels 60+ and now, blizzard decided to completely change all the stat conversion rates for all levels.

Since I use a hard coded table for that data (since I can't find a way to dynamically pull it from the game) when they changed all those numbers, my data didn't update and as such resulted in incorrect calculations.

I did a quick local test and it seems to have fixed it (at least from what I can tell) so I will just have to update all those conversion rates and then it should work correctly .......

In theory.
